CAIPs/caip-x.md

```diff
@@ -0,0 +1,222 @@
+discussions-to: https://github.com/ChainAgnostic/CAIPs/pull/275
+status: Draft
+type: Standards Track
+Standard created: 2024-04-29
+updated: 2024-04-29
+---
+
+# Abstract
+
+The Domain Wallet Authentication describes a method for linking a crypto domain with authentication methods or providers by adding an authenticator: JSON/URL field to the metadata of a crypto domain NFT. The standard also describes a method for application developers and web3 login modal providers to enable users to login with their domain name.
+
+The goal of this specification is to define a chain-agnostic identity standard for domain-based authentication.
+
+# Motivation
+
+Current blockchain authentication methods primarily rely on connecting wallets via providers like Metamask. However, this requires users to remember both their wallet provider and their wallet address.  As more wallets, signers, and chains come online, this problem will only get worse.
+
+Crypto domains provide a human-readable, user-friendly way to represent wallet addresses. By enabling authentication directly with crypto domains, this standard aims to improve usability and adoption of web3 logins.
+
+Additionally, standardizing the way domain NFT metadata specifies its associated authentication provider allows any compatible domain NFT to become a domain wallet or any key management or authentication method. Additionally, any login modal or dApp can easily integrate domain-based logins.
+
+# Specification
+
+## Storage Format
+
+Any system capable of resolving text records can be used for the name in this system.  However, we have chosen to focus this specification on Crypto domain NFTs.  Crypto domain NFTs that are compatible with this authentication standard MUST include an authenticator text record entry with the following properties:
+authenticator (string, required): A URL that points to a JSON configuration where the app can find out how to auth the user.  e.g. http://www.authprovider.com/auth/{}
+The application will craft the final URL to get the configuration, where "{}" will be substituted for the user's whole crypto domain name, so for "chrisc.eth" the final URL would be "http://www.authprovider.com/auth/chrisc.eth"
+
+The actual standard used to store this "authenticator" text record will vary depending on the Crypto Domain NFT system used.  For example, on ENS, a ENSIP-5 record should be created, with a "key" of "authenticator".  You can lookup the authenticator on an ENS domain using the text(bytes32 node, string key) function where "node" is the ENS domain being queried and "key" is the string "authenticator".
+
+Example of a response to retrieve the "authenticator" text record:
+
+"https://www.authprovider.com/auth/{}"
+
+## Authentication flow definition
+
+The User can provide their authentication flow by providing a URL where Web3 applications can fetch it.
```

Comment:
I see your point, and agree. Most users won't ever deal with this

I believe the most common use case for this will be wallets that define their own address naming system and integrate with ENS, such as coinbase domains under `*.cb.id`. They provide naming, and could also provide `authenticator` text records, offline

The self hosting option or the feature to write it yourself in the `authenticator` text record still have to be there as those are the only options that would allow any user to not trust anybody and to have as much freedom as they want/need (maybe combining competing wallets)

Anyway we rephrased this section, hopefully it is more clear now what we wanted to express